cloudferrylib/base/resource.py

```diff
@@ -36,8 +41,18 @@ def save(self):
     def restore(self):
         pass
 
-    def wait_for_status(self, id_obj, status, limit_retry=60):
-        pass
+    def wait_for_status(self, res_id, get_status, wait_status, limit_retry=60, warn_only=False):
```

Comment:
#run pep checker
find . -name '*.py' -not -wholename './.venv/*' | xargs .venv/bin/flake8 --max-line-length 160 || true